reactive/coredns.py

```diff
@@ -1,22 +1,57 @@
+from pathlib import Path
+
 from charmhelpers.core import hookenv
-from charms.reactive import set_flag, clear_flag
+from charms.reactive import set_flag, clear_flag, endpoint_from_flag
 from charms.reactive import when, when_not
 
 from charms import layer
 
 
+@when_not('cluster-dns.joined')
+def wait_for_cluster_dns():
+    layer.status.maintenance('Waiting for cluster dns relation')
+
+
 @when('layer.docker-resource.coredns-image.available')
 @when_not('charm.coredns.started')
+@when('cluster-dns.domain-ready')
```

Comment:
It looks like the cluster-dns relation is required, not optional. Do we intend for this charm to be usable on other Kubernetes distributions, e.g. microk8s? If so, then this requirement will be a problem, right?

Reply:
It needs the dns_domain when deploying the pods in the beginning. It could use a default domain name and redeploy everything once it recognizes the cluster-dns relation being ready. Although I'm not entirely sure how important it is that this works for other Kubernetes distributions.